cyhy/core/geoloc.py

```diff
@@ -39,3 +41,19 @@ def lookup(self, ip):
             return (response.location.longitude, response.location.latitude)
         except AddressNotFoundError:
             return (None, None)
+
+    def check_restricted_cidr(self, cidr):
+        restricted_ips = IPSet()
+        for ip in cidr:
+            try:
+                response = self.__reader.city(str(ip))
+                if response.country.name in RESTRICTED_COUNTRIES:
+                    restricted_ips.add(ip)
+            except AddressNotFoundError:
+                print >> sys.stderr, "CIDR %s not found in geolocation database" % cidr
+        if len(restricted_ips) > 0:
+            print >> sys.stderr, "Warning! The following networks can be traced to restricted country: %s" % response.country.name
+            for network in restricted_ips.iter_cidrs():
+                print >> sys.stderr, network
+            return True
+        return False
```

Comment:
```suggestion
    def check_restricted_ip(self, ip):
        try:
            response = self.__reader.city(str(ip))
            if response.country.name in RESTRICTED_COUNTRIES:
                return True
            return False
        except AddressNotFoundError:
            print >> sys.stderr, "CIDR %s not found in geolocation database" % cidr
```

Reply:
I will commit this suggestion, but will slightly alter it in a different commit. The wording should be changed to indicate the IP is not found in the geoloc db instead of the cidr.

Reply:
Actually, thinking about it more... one thing I did like about the current implementation is the ability to point out which country it was traced to. If that is not necessary, then I won't worry about it, but if we do want that functionality, I could have it return the country as well as the true/false possibly, but that might be a little clunky. Bottom line though, is do we care about knowing which country it is from or do we only care that it is in the list?

Reply:
I'd be ok with returning a tuple of

```python
(boolean, "Country name or empty string")
```

if you wanted that information. You could do a dict of `"Country Name": IPSet()` key/values in the scripts to minimize output.

Reply:
Additional thought I had is you could consider using [`namedtuple()`](https://docs.python.org/2.7/library/collections.html#collections.namedtuple) to manage the return value. I use these when I want to provide a return that contains multiple values.